src/webportal/src/app/home/index.jsx

```diff
@@ -0,0 +1,259 @@
+      >
+        Learn more
+      </Link>
+    </div>
+    <div className={c(t.w33, t.w100M, t.mt5M, t.tc, t.flex, t.flexColumn, t.itemsCenter, t.justifyBetween)}>
+      <div className={c(t.flex, t.flexColumn, t.itemsCenter)}>
+        <div className={c(FontClassNames.xxLarge)} style={{fontWeight: FontWeights.semibold}}>
+          Use VS Code Extension to work with Jobs
+        </div>
+        <div className={c(FontClassNames.mediumPlus, t.lhCopy, t.mv4)} style={{maxWidth: '20rem'}}>
+          OpenPAI Client is a VS Code extension to connect PAI clusters, submit AI jobs, and manage files on HDFS, etc. You need to install the extension in VS code before using it.
+        </div>
+      </div>
+      <Link
+        href='https://github.com/Microsoft/pai/blob/master/contrib/pai_vscode/VSCodeExt.md'
+        target='_blank'
+        style={{fontWeight: FontWeights.semibold}}
+      >
+        Learn more
+      </Link>
+    </div>
+  </div>
+);
+
+async function login(username, password, expiration = 7 * 24 * 60 * 60) {
+  const res = await fetch(`${config.restServerUri}/api/v1/token`, {
+    method: 'POST',
+    body: JSON.stringify({
+      username,
+      password,
+      expiration,
+    }),
+    headers: {
+      'content-type': 'application/json',
+    },
+  });
+  if (res.ok) {
+    const data = await res.json();
+    if (data.error) {
+      throw new Error(data.message);
+    } else {
+      cookies.set('user', data.user, {expires: expiration});
+      cookies.set('token', data.token, {expires: expiration});
+      cookies.set('admin', data.admin, {expires: expiration});
+      cookies.set('hasGitHubPAT', data.hasGitHubPAT, {expires: expiration});
+    }
+  } else {
+    const data = await res.json();
+    throw new Error(data.message);
+  }
+}
+
+const Home = () => {
+  const [loginModal, setLoginModal] = useState(false);
+  const [error, setError] = useState(null);
+  const [lock, setLock] = useState(false);
+  const usernameRef = useRef(null);
+  const passwordRef = useRef(null);
+  const onLoginClick = () => {
+    setLock(true);
+    void login(
+      usernameRef.current.value,
+      passwordRef.current.value
+    ).then(() => {
+      window.location.replace(loginTarget);
+    }).catch((e) => {
+      setError(e.message);
+    }).finally(() => {
+      setLock(false);
+    });
+  };
+
+  return (
+    <div className={c(t.minVh100, t.w100, t.flex, t.flexColumn)}>
+      {/* top */}
+      <div className={c(t.bgBlack, t.pv3, t.ph4, t.flex, t.justifyBetween)}>
+        <div className={c(FontClassNames.large, t.white)}>
+          Platform for AI
+        </div>
+        <div className={c(FontClassNames.large, t.white, t.dim, t.pointer)} onClick={() => setLoginModal(true)}>
+          Sign in
+        </div>
+      </div>
+      {/* content */}
+      <div className={c(t.flexAuto, t.flex, t.flexColumn, t.relative)}>
+        {/* jumbotron */}
+        <Jumbotron showLoginModal={() => setLoginModal(true)} />
```

Comment:
It's better to wrap the callback with `useCallback` to cache it

Reply:
fixed